Spark kernel fails if any Additional JVM arguments are specified in the notebook configuration

**Description**
Spark kernel fails if any Additional JVM arguments are specified in the notebook configuration. It does not make any difference what parameters you pass. The presence of any arg causes the issue.

**Reproduction**
Steps to reproduce the behavior:
1. Create a mew Spark-enabled notebook
2. Open the notebook configuration and add any JVM arg to the "Additional JVM arguments" text box. `-agentlib:jdwp=transport=dt_socket,server=y,address=8000,suspend=y`, for example. But it can be anything.
3. Click Save and Restart.
4. Enjoy the "Kernel died" error on UI and a bunch of exceptions in the logs. The actual problem is that the arguments are somehow wrapped with List() - ` List(-agentlib:jdwp=transport=dt_socket,server=y,address=8000,suspend=y)`. See the log below.

```
server_1  |    |     spark-submit --class polynote.kernel.remote.RemoteKernelClient --name "Polynote 0.4.2: test3.ipynb" --driver-java-options "List(-agentlib:jdwp=transport=dt_socket,server=y,address=8000,suspend=y) -Dlog4j.configuration=log4j.properties -Djava.library.path=/usr/local/lib/python3.7/dist-packages/jep" --driver-class-path /opt/polynote/deps/2.12/polynote-spark-runtime-assembly-0.4.2.jar:/opt/polynote/deps/2.12/scala-collection-compat_2.12-2.1.1.jar:/opt/polynote/deps/2.12/polynote-server-assembly-0.4.2.jar:/opt/polynote/deps/2.12/polynote-runtime-assembly-0.4.2.jar:/opt/polynote/deps/2.12/polynote-kernel-assembly-0.4.2.jar:/opt/polynote/deps/2.12/scala-library-2.12.12.jar:/opt/polynote/deps/2.12/scala-xml_2.12-1.2.0.jar:/opt/polynote/deps/2.12/scala-compiler-2.12.12.jar:/opt/polynote/deps/2.12/polynote-spark-assembly-0.4.2.jar:/opt/polynote/deps/2.12/scala-reflect-2.12.12.jar --jars file:/opt/polynote/deps/2.12/polynote-spark-runtime-assembly-0.4.2.jar,file:/opt/polynote/deps/2.12/polynote-runtime-assembly-0.4.2.jar --conf spark.master=local[*] /opt/polynote/deps/2.12/polynote-spark-assembly-0.4.2.jar --address 127.0.0.1 --port 44251 --kernelFactory polynote.kernel.LocalSparkKernelFactory
server_1  | [REMOTE | test3.ipynb]
server_1  |         |   Error: Could not find or load main class List(-agentlib:jdwp=transport=dt_socket,server=y,address=8000,suspend=y)
server_1  | [INFO]   Kernel process ended with 1
server_1  | [ERROR]  Error starting kernel; shutting it down (Logged from KernelPublisher.scala:110)
server_1  |    |     polynote.kernel.remote.SocketTransport$ProcessDied$: Kernel died unexpectedly
server_1  |    |     polynote.kernel.remote.SocketTransport$ProcessDied$.<clinit>(transport.scala)
server_1  |    |     polynote.kernel.remote.SocketTransportServer$.$anonfun$monitorProcess$9(transport.scala:150)
server_1  |    |     zio.ZIO$.$anonfun$fail$1(ZIO.scala:2733)
server_1  |    |     zio.internal.FiberContext.evaluateNow(FiberContext.scala:407)
server_1  |    |     zio.internal.FiberContext.$anonfun$evaluateLater$1(FiberContext.scala:775)
server_1  |    |     java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
server_1  |    |     java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
server_1  |    |     java.lang.Thread.run(Thread.java:748)
server_1  | [ERROR]   (Logged from NotebookSession.scala:193)
server_1  |    |     polynote.kernel.remote.SocketTransport$ProcessDied$: Kernel died unexpectedly
server_1  |    |     polynote.kernel.remote.SocketTransport$ProcessDied$.<clinit>(transport.scala)
server_1  |    |     polynote.kernel.remote.SocketTransportServer$.$anonfun$monitorProcess$9(transport.scala:150)
server_1  |    |     zio.ZIO$.$anonfun$fail$1(ZIO.scala:2733)
server_1  |    |     zio.internal.FiberContext.evaluateNow(FiberContext.scala:407)
server_1  |    |     zio.internal.FiberContext.$anonfun$evaluateLater$1(FiberContext.scala:775)
server_1  |    |     java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
server_1  |    |     java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
server_1  |    |     java.lang.Thread.run(Thread.java:748)
```

- Polynote Version: 
   - 0.4.2

--- a/polynote-kernel/src/main/scala/polynote/kernel/remote/DeploySparkSubmit.scala
+++ b/polynote-kernel/src/main/scala/polynote/kernel/remote/DeploySparkSubmit.scala
@@ -48,10 +48,11 @@ object DeploySparkSubmit extends DeployCommand {
     val isRemote = sparkConfig.get("spark.submit.deployMode") contains "cluster"
 
     val allDriverOptions = {
-      nbConfig.jvmArgs.toList ++
-      sparkConfig.get("spark.driver.extraJavaOptions").toList ++
-      asPropString(javaOptions)
-    } mkString " "
+      val all: List[String] = jvmArgs(nbConfig) ++
+        sparkConfig.get("spark.driver.extraJavaOptions").toList ++
+        asPropString(javaOptions)
+      all mkString " "
+    }
 
     val additionalJars = classPath.toList.filter(_.getFile.endsWith(".jar"))
 

--- a/polynote-kernel/src/main/scala/polynote/kernel/remote/package.scala
+++ b/polynote-kernel/src/main/scala/polynote/kernel/remote/package.scala
@@ -1,6 +1,7 @@
 package polynote.kernel
 
 import polynote.kernel.util.RPublish
+import polynote.messages.NotebookConfig
 import zio.Has
 
 import java.io.File
@@ -24,4 +25,5 @@ package object remote {
     case (name, value) => s"-D$name=$value"
   }
 
+  def jvmArgs(nbConfig: NotebookConfig) = nbConfig.jvmArgs.toList.flatten
 }

--- a/polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala
+++ b/polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala
@@ -425,7 +425,7 @@ object SocketTransport {
             .filterNot(_.isEmpty)
             .mkString(File.pathSeparator)
 
-          val javaArgs = notebookConfig.jvmArgs.toList.flatten ++ asPropString(javaOptions)
+          val javaArgs = jvmArgs(notebookConfig) ++ asPropString(javaOptions)
 
           java :: "-cp" :: fullClassPath :: javaArgs :::
             classOf[RemoteKernelClient].getName ::

--- a/polynote-kernel/src/test/scala/polynote/kernel/remote/DeploySparkSubmitSpec.scala
+++ b/polynote-kernel/src/test/scala/polynote/kernel/remote/DeploySparkSubmitSpec.scala
@@ -1,14 +1,44 @@
 package polynote.kernel.remote
 
 import org.scalatest.{FreeSpec, Matchers}
+import polynote.config.PolynoteConfig
+import polynote.buildinfo.BuildInfo
+import polynote.messages.{NotebookConfig, TinyList}
 import polynote.testing.ZIOSpec
 import polynote.testing.MockSystem
 import zio.ZIO
 
 import java.io.File
+import java.net.{URI, URL}
 import java.nio.file.{Files, Paths}
 
 class DeploySparkSubmitSpec extends FreeSpec with Matchers with ZIOSpec {
+
+  "Generates a proper command string" in {
+    val conf = PolynoteConfig()
+    val jvmProps: Option[TinyList[String]] = Option(List("-Dprop=value"))
+    val nbConf = NotebookConfig(None, None, None, None, None, None, jvmArgs = jvmProps)
+    val nbPath = "foo"
+    val assemblyJar = "/path/to/polynote/polynote-spark-assembly.jar"
+    val cp = Seq(
+      URI.create("file:///path/to/my.jar").toURL,
+      URI.create("file:///path/to/not-a-jar/").toURL,
+      URI.create(s"file://${assemblyJar}").toURL)
+    val serverArgs = List("--address", "blah", "--port", "12345")
+
+    val expected =
+      "spark-submit" ::
+        "--class" :: "polynote.kernel.remote.RemoteKernelClient" ::
+        "--name" :: s"Polynote ${BuildInfo.version}: $nbPath" ::
+        "--driver-java-options" :: (jvmArgs(nbConf) ++ asPropString(javaOptions)).mkString(" ") ::
+        "--driver-class-path" :: cp.map(_.getPath).mkString(File.pathSeparator) ::
+        assemblyJar ::
+        serverArgs
+
+    val command = DeploySparkSubmit.build(conf, nbConf, nbPath, cp, serverArgs = serverArgs)
+
+    command should contain theSameElementsInOrderAs expected
+  }
 
   "Detect scala version" - {
 

